docs/xks/developer-guide/container-security.md

```diff
@@ -0,0 +1,293 @@
+
+Of course there is a solution for applications that still do want to write to the filesystem. The solution is to create a [`emptyDir`](https://kubernetes.io/docs/concepts/storage/volumes/#emptydir) volume and mount it to the directory to which files should be written. In the example below the `emptyDir` volume `tmp` is mounted to the path `/tmp` making it possible to write files to the directory.
+
+```yaml
+apiVersion: v1
+kind: Pod
+metadata:
+  name: nginx
+spec:
+  containers:
+    - name: nginx
+      image: nginx:1.14.2
+      securityContext:
+        readOnlyRootFilesystem: true
+        allowPrivilegeEscalation: false
+        capabilities:
+          drop:
+            - NET_RAW
+            - CAP_SYS_ADMIN
+      volumeMounts:
+        - mountPath: /tmp
+          name: tmp
+  volumes:
+    - emptyDir: {}
+      name: tmp
+```
+
+### Allow Privilege Escalation
+
+Disabling privilege escalation for all Pods is critical for applications to to gain greater access to the Node than they need to. This setting will probably never affect most applications.
+
+### Dropped Capabilities
+
+Linux capabilities are certain root permissions that can be given to containers without giving root access. Certain capabilities should never be given to a container as they would give too much access, for this reason they are always dropped.
+
+* `CAP_SYS_ADMIN` - Is the same as making the container root, would allow a container to escalate privileges.
+* `NET_RAW` - Allows a container to craft raw network packets which can be exploited for malicious actions.
+
+## Pod Constraints
+
+In addition to the default security context there are certain configuration settings which are not allowed in XKF. They have a lower impact on end users as they are opted into, but they are good to be aware of and can explain why a Pods creation may be blocked by the cluster. The enforcement is done before the Pod is created, this means that a Deployment may be allowed to be created but the Pods that result will not, causing a Deployment to always have zero replicas.
+
+### Host Namespaces
+
+Being able to share process IDs or memory between a container and a host means that the container would in theory have access to the host or any containers also running on the host. Setting `hostPID` or `hostIPC` to `true` would allow for this to happen. For this reason that is not allowed.
+
+```yaml
+apiVersion: v1
+kind: Pod
+metadata:
+  name: app
+spec:
+  hostPID: true
+  hostIPC: true
```

Comment:
You say this is not allowed, yet your examples above say `true` whereas in other examples you show the actual config. Did you mean `false` here?

(see further examples of similar issues below)